fluss-client/src/main/java/com/alibaba/fluss/client/admin/Admin.java

```diff
@@ -247,6 +247,23 @@ CompletableFuture<Void> createTable(
      */
     CompletableFuture<List<PartitionInfo>> listPartitionInfos(TablePath tablePath);
 
+    /**
+     * List all partitions in the given table in fluss cluster asynchronously.
+     *
+     * <p>The following exceptions can be anticipated when calling {@code get()} on returned future.
+     *
+     * <ul>
+     *   <li>{@link TableNotExistException} if the table does not exist.
+     *   <li>{@link TableNotPartitionedException} if the table is not partitioned.
+     *   <li>{@link InvalidPartitionException} if the input partition spec is invalid.
+     * </ul>
+     *
+     * @param tablePath The path of the table.
+     * @param partitionSpec The partition spec of the table
+     */
+    CompletableFuture<List<PartitionInfo>> listPartitionInfos(
+            TablePath tablePath, PartitionSpec partitionSpec);
```

Comment:
`partitionSpec` -> `partialPartitionSpec`

Add descrption about the `partialPartitionSpec` and explain the difference between this method and `listPartitionInfos(TablePath tablePath)`.

You can take the Javadoc of `org.apache.flink.table.catalog.Catalog#listPartitions(org.apache.flink.table.catalog.ObjectPath, org.apache.flink.table.catalog.CatalogPartitionSpec)`  as an example.